### ISSUE TITLE
Feature - Check if device is compatible through eureka endpoint

### ISSUE THREAD

**Issue by DurgNomis-drol** [leikoilja/ha-google-home#220]
Actually even if the devices returns a `200` for `eureka_info`, there is no guaranty that the device supports alarms or timers, we still would need to check `assistant_supported`. This can also come in handy in the future where we can check what other features is supported. 

The check should probably be a combination of both like this:

1. Request `eureka_info` if `404` return device **not** compatible. Else continue check.
2. Check if `assistant_supported` is true, return device is compatible. If false return device is **not** compatible.

_Originally posted by @DurgNomis-drol in https://github.com/leikoilja/ha-google-home/issues/212#issuecomment-821975322_

**Comment by KapJI** [comment]
If `eureka_info` is 200, we should create `device` sensor. It `assistant_supported` is true we should also create alarms and timers sensors and do not disturb switch.

**Comment by leikoilja** [comment]
Great idea! That exactly what i was thinking in [earlier discussions](https://github.com/leikoilja/ha-google-home/pull/133#issue-607879223), we could probably take a step backwards and implement it in the following manner:
- Request devices from foyer (glocaltokens)
- Initialize `GoogleHomeDevice` as we currently do in `model` per device 
- Do `eureka` call to populate `GoogleHomeDevice` objects with necessary information (like `capabilities`) etc. (that can be expanded over time i guess
- Initialize HA sensors/services based on those object arguments stated in `GoogleHomeDevice` objects

**Comment by stboch** [comment]
yup also the reboot device function can be based on reboot_supported. A stub for future Bluetooth work based on ble_supported.
